runtime/lua/vim/lsp/buf.lua

```diff
@@ -136,6 +136,39 @@ function M.document_symbol()
   request('textDocument/documentSymbol', params)
 end
 
+local function pick_call_hierarchy_item(call_hierarchy_items)
+  if not call_hierarchy_items then return end
+  if #call_hierarchy_items == 1 then
+    return call_hierarchy_items[1]
+  end
+  local items = {}
+  for i, item in ipairs(call_hierarchy_items) do
+    local entry = item.detail or item.name
+    table.insert(items, string.format("%d. %s", i, entry))
+  end
+  local choice = vim.fn.inputlist(items)
+  if choice < 1 or choice > #items then
+    return
+  end
+  return choice
+end
+
+function M.incoming_calls()
+  local params = util.make_position_params()
+  request('textDocument/prepareCallHierarchy', params, function(_, _, result)
```

Comment:
How will this work with multiple language servers?

Reply:
What do you mean exactly?

Reply:
There may be multiple servers running for a given language. Currently neovim sends the request to all servers and you get a list of responses in the callback (most methods look at the first result), I think.

Reply:
Sorry nevermind you're already handling it in the capability check (if the user has multiple servers with repeated capabilities it's kidna hard to know what to do hehe)

Reply:
Should this be a synchronous request?

Reply:
Probably not, getting the list of all calls can easily take over 10 seconds on large projects in CLion and it might be worse for some servers.